distribution/pull_v2.go

```diff
@@ -861,6 +871,69 @@ func fixManifestLayers(m *schema1.Manifest) error {
 	return nil
 }
 
-func createDownloadFile() (*os.File, error) {
-	return ioutil.TempFile("", "GetImageBlob")
+var v2LayerDataFilename = "v2data"
+
+type resumeData struct {
+	offset   int64
+	file     *os.File
+	verifier digest.Verifier
+}
+
+func resumeOrCreateDownloadFile(progressOutput progress.Output, cacheDir string, dgst digest.Digest, id string) (resumeData, error) {
+	dataPath := filepath.Join(cacheDir, v2LayerDataFilename)
+	_, err := os.Stat(dataPath)
+	if err == nil {
+		return resumeDownloadFile(progressOutput, cacheDir, dgst, id)
+	}
+	if os.IsNotExist(err) {
+		tmpFile, err := createDownloadFile(cacheDir)
+		if err != nil {
+			return resumeData{}, err
+		}
+		return resumeData{file: tmpFile}, nil
```

Comment:
Could just do `return resumeData{file: tmpFile}, err` here and skip checking `err`, but it doesn't really matter either way.